<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sean Wilson" />

<meta name="date" content="2020-11-12" />

<title>BulkRNA-Seq</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HowdenWilson2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="HFK_Annotation.html">HFK</a>
</li>
<li>
  <a href="BulkRNA-Seq.html">Bulk</a>
</li>
<li>
  <a href="Figures.html">Figures</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sbwilson91/HowdenWilson2020">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">BulkRNA-Seq</h1>
<h4 class="author">Sean Wilson</h4>
<h4 class="date">2020-11-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HowdenWilson2020/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201109code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201109code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201109)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKidneyRegenerationHowdenWilson2020tree11477ded30366e99ed3566c3ae910e3fb4c1f47atargetblank11477dea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KidneyRegeneration/HowdenWilson2020/tree/11477ded30366e99ed3566c3ae910e3fb4c1f47a" target="_blank">11477de</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKidneyRegenerationHowdenWilson2020tree11477ded30366e99ed3566c3ae910e3fb4c1f47atargetblank11477dea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KidneyRegeneration/HowdenWilson2020/tree/11477ded30366e99ed3566c3ae910e3fb4c1f47a" target="_blank">11477de</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  README.html
    Untracked:  code/AddModuleScore_bulk.R
    Untracked:  code/GeneSets.Rmd
    Untracked:  code/functions.R
    Untracked:  code/hfk.R
    Untracked:  code/job_identity_markers.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BulkRNA-Seq.Rmd</code>) and HTML (<code>docs/BulkRNA-Seq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/11477ded30366e99ed3566c3ae910e3fb4c1f47a/analysis/BulkRNA-Seq.Rmd" target="_blank">11477de</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
<td>
Updating additional information around analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KidneyRegeneration/HowdenWilson2020/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/BulkRNA-Seq.html" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KidneyRegeneration/HowdenWilson2020/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/BulkRNA-Seq.html" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/711a2c5d65bfe450a0c6870effe2fd16a4e4ea77/analysis/BulkRNA-Seq.Rmd" target="_blank">711a2c5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
<td>
Set up initial repo on github
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We generated a number of Bulk RNA-Seq samples to analyse the transcriptomes of cell populations and cultures from this study. <br> <br> In the first instance, we used a GATA3:mCherry reporter iPS cell line to generate organoids. These organoids were dissociated and stained with a conjugated EPCAM-Fluor488 antibody before being FACS sorted. We collected the various populations and Bulk RNA-Sequenced them. Ureteric Epithelium cultures generated from the GATA3+/EPCAM+ cells cultured in UE supporting conditions were also collected and sequenced.<br> <br> These samples are “batch 1”.</p>
<pre><code>Loading required package: limma</code></pre>
<pre><code>Loading required package: pkgmaker</code></pre>
<pre><code>Loading required package: registry</code></pre>
<pre><code>Loading required package: rngtools</code></pre>
<pre><code>Loading required package: cluster</code></pre>
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: synchronicity] | Cores 31/32</code></pre>
<pre><code>  To enable shared memory capabilities, try: install.extras(&#39;
NMF
&#39;)</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:NMF&#39;:

    nrun</code></pre>
<pre><code>The following object is masked from &#39;package:pkgmaker&#39;:

    new2</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: OrganismDbi</code></pre>
<pre><code>Loading required package: GenomicFeatures</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GO.db</code></pre>
<pre><code></code></pre>
<pre><code>Loading required package: org.Hs.eg.db</code></pre>
<pre><code></code></pre>
<pre><code>Loading required package: TxDb.Hsapiens.UCSC.hg19.knownGene</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    space</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    space</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre><code>Registered S3 method overwritten by &#39;cli&#39;:
  method     from    
  print.boxx spatstat</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.0     ✓ purrr   0.3.4
✓ tibble  3.0.4     ✓ dplyr   1.0.2
✓ tidyr   1.0.2     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::collapse()   masks IRanges::collapse()
x dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()
x dplyr::desc()       masks IRanges::desc()
x tidyr::expand()     masks S4Vectors::expand()
x dplyr::filter()     masks stats::filter()
x dplyr::first()      masks S4Vectors::first()
x dplyr::lag()        masks stats::lag()
x ggplot2::Position() masks BiocGenerics::Position(), base::Position()
x purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()
x dplyr::rename()     masks S4Vectors::rename()
x dplyr::select()     masks OrganismDbi::select(), AnnotationDbi::select()
x dplyr::slice()      masks IRanges::slice()</code></pre>
</div>
<div id="preparing-the-data" class="section level2">
<h2>Preparing the data</h2>
<pre><code>                  ID      Group
1       UEculture_r1  UEculture
2       UEculture_r2  UEculture
3       UEculture_r3  UEculture
4   Org_mCh_EPCAM_r1  mCh_EPCAM
5   Org_mCh_EPCAM_r2  mCh_EPCAM
6   Org_mCh_EPCAM_r3  mCh_EPCAM
7    Org_mCh_only_r1   mCh_only
8    Org_mCh_only_r2   mCh_only
9    Org_mCh_only_r3   mCh_only
10 Org_EPCAM_only_r1 EPCAM_only
11 Org_EPCAM_only_r2 EPCAM_only
12 Org_EPCAM_only_r3 EPCAM_only</code></pre>
<p>Create a DGElist object and add gene annotation information to the data.</p>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<p>Visualise library sizes</p>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-3-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Remove genes with less than 1 count per million in at least 3 samples. Then normalise using TMM.</p>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-5-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 500  12</code></pre>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-6-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="statistical-analysis" class="section level2">
<h2>Statistical analysis</h2>
<p>The differential expression analysis is performed using <code>voom</code> and <em>limma</em>. The <code>voom</code> (Law et al. 2014) function is used on the normalized counts to robustly estimate the mean-variance relationship and generate precision weights for each observation. Gene-wise linear models are then fitted to the voom-transformed log2 CPM to determine differences in gene expression.</p>
<p>Statistically significant differentially expressed genes are identified using empirical Bayes moderated t-tests (Smyth 2005), allowing for a mean-variance trend and performing robust empirical Bayes shrinkage of the gene-wise variances to protect against hypervariable genes (Phipson, Lee, et al. 2016). P-values are adjusted for false discovery rate using the Benjamini-Hochberg method (Benjamini &amp; Hochberg 1995).</p>
<p>Set up the design matrix and apply voom to the counts.</p>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-7-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Fit the linear model.</p>
<p>Summarise the numbers of differentially expressed genes for each comparison.</p>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-9-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As there are many differentially expressed genes in the original analysis, we will also test relative to a threshold (TREAT) of absolute log2 fold change &gt; 1.</p>
<p>Summarise the numbers of differentially expressed genes for each comparison. There are fewer DE genes from the TREAT analysis.</p>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/84370d51eb1680b0cd69730aeaf9ee087e059b0c/docs/figure/BulkRNA-Seq.Rmd/unnamed-chunk-11-1.png" target="_blank">84370d5</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-11-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-set-scores" class="section level2">
<h2>Gene set scores</h2>
<p>We generated from the human fetal kidney single cell data Gene Sets that we could use to identify each population within the HFK and ask how similar each sample. The Gene Sets are derived from stringent DGE tests between each individual cluster and finding sets of genes that are most representative for each cluster identity.<br> These are generated in the below script.</p>
<p>Then we can visualise how these look on a pseudobulk analysis of the original HFK data.</p>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre><code>[1] &quot;95 N.DCT&quot;
[1] &quot;28 N.NPC_Primed&quot;
[1] &quot;165 N.NPC_CC&quot;
[1] &quot;68 N.NPC&quot;
[1] &quot;24 N.Distal_EN&quot;
[1] &quot;170 S.IC&quot;
[1] &quot;228 U.Med_Inner&quot;
[1] &quot;216 S.Med&quot;
[1] &quot;548 Endothelium&quot;
[1] &quot;119 N.CS&quot;
[1] &quot;406 N.Pod_Mat&quot;
[1] &quot;216 N.Pod_Dev&quot;
[1] &quot;47 N.Medial_EN&quot;
[1] &quot;134 U.Cortical&quot;
[1] &quot;47 N.RV&quot;
[1] &quot;133 S.OC_NZ&quot;
[1] &quot;113 N.PEC&quot;
[1] &quot;27 N.DST&quot;
[1] &quot;11 S.NP_Str&quot;
[1] &quot;42 N.NP_Str&quot;
[1] &quot;205 N.PT_Dev&quot;
[1] &quot;117 U.Med_Outer&quot;
[1] &quot;31 N.NPC_PTA&quot;
[1] &quot;385 N.PT_Mat&quot;
[1] &quot;72 U.Tip&quot;
[1] &quot;231 N.NPC_PTA_CC&quot;
[1] &quot;710 Immune&quot;
[1] &quot;156 N.LoH&quot;
[1] &quot;183 S.Mesangial&quot;</code></pre>
<p><img src="figure/BulkRNA-Seq.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /hpc/software/installed/R/3.6.1/lib64/R/lib/libRblas.so
LAPACK: /hpc/software/installed/R/3.6.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.0                          
 [2] stringr_1.4.0                          
 [3] dplyr_1.0.2                            
 [4] purrr_0.3.4                            
 [5] readr_1.4.0                            
 [6] tidyr_1.0.2                            
 [7] tibble_3.0.4                           
 [8] ggplot2_3.3.0                          
 [9] tidyverse_1.3.0                        
[10] gplots_3.1.0                           
[11] Glimma_1.14.0                          
[12] Homo.sapiens_1.3.1                     
[13] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[14] org.Hs.eg.db_3.8.2                     
[15] GO.db_3.8.2                            
[16] OrganismDbi_1.28.0                     
[17] GenomicFeatures_1.36.4                 
[18] GenomicRanges_1.36.1                   
[19] GenomeInfoDb_1.20.0                    
[20] AnnotationDbi_1.46.1                   
[21] IRanges_2.20.2                         
[22] S4Vectors_0.24.3                       
[23] NMF_0.22.0                             
[24] bigmemory_4.5.36                       
[25] Biobase_2.46.0                         
[26] BiocGenerics_0.32.0                    
[27] cluster_2.1.0                          
[28] rngtools_1.5                           
[29] pkgmaker_0.32.2                        
[30] registry_0.5-1                         
[31] RColorBrewer_1.1-2                     
[32] Seurat_3.2.2                           
[33] edgeR_3.28.1                           
[34] limma_3.42.2                           
[35] workflowr_1.6.2                        

loaded via a namespace (and not attached):
  [1] reticulate_1.16             tidyselect_1.1.0           
  [3] RSQLite_2.2.1               htmlwidgets_1.5.1          
  [5] grid_3.6.1                  BiocParallel_1.20.1        
  [7] Rtsne_0.15                  munsell_0.5.0              
  [9] codetools_0.2-16            ica_1.0-2                  
 [11] statmod_1.4.35              future_1.16.0              
 [13] miniUI_0.1.1.1              withr_2.2.0                
 [15] colorspace_1.4-1            knitr_1.28                 
 [17] rstudioapi_0.11             ROCR_1.0-11                
 [19] tensor_1.5                  listenv_0.8.0              
 [21] git2r_0.27.1                GenomeInfoDbData_1.2.1     
 [23] polyclip_1.10-0             bit64_4.0.5                
 [25] rprojroot_1.3-2             vctrs_0.3.4                
 [27] generics_0.1.0              xfun_0.12                  
 [29] R6_2.5.0                    doParallel_1.0.16          
 [31] rsvd_1.0.3                  locfit_1.5-9.4             
 [33] bitops_1.0-6                spatstat.utils_1.17-0      
 [35] DelayedArray_0.12.2         assertthat_0.2.1           
 [37] promises_1.1.0              scales_1.1.1               
 [39] gtable_0.3.0                globals_0.13.0             
 [41] goftest_1.2-2               rlang_0.4.7                
 [43] splines_3.6.1               rtracklayer_1.44.4         
 [45] lazyeval_0.2.2              broom_0.7.2                
 [47] modelr_0.1.6                BiocManager_1.30.10        
 [49] yaml_2.2.1                  reshape2_1.4.4             
 [51] abind_1.4-5                 backports_1.1.10           
 [53] httpuv_1.5.2                RBGL_1.62.1                
 [55] tools_3.6.1                 gridBase_0.4-7             
 [57] ellipsis_0.3.1              ggridges_0.5.2             
 [59] Rcpp_1.0.5                  plyr_1.8.6                 
 [61] progress_1.2.2              zlibbioc_1.32.0            
 [63] RCurl_1.98-1.2              prettyunits_1.1.1          
 [65] rpart_4.1-15                deldir_0.1-29              
 [67] pbapply_1.4-3               cowplot_1.1.0              
 [69] zoo_1.8-8                   haven_2.2.0                
 [71] SummarizedExperiment_1.16.1 ggrepel_0.8.2              
 [73] fs_1.5.0                    here_0.1                   
 [75] magrittr_1.5                data.table_1.13.0          
 [77] reprex_0.3.0                lmtest_0.9-38              
 [79] RANN_2.6.1                  whisker_0.4                
 [81] fitdistrplus_1.1-1          matrixStats_0.57.0         
 [83] hms_0.5.3                   patchwork_1.0.0            
 [85] mime_0.9                    evaluate_0.14              
 [87] xtable_1.8-4                XML_3.99-0.3               
 [89] readxl_1.3.1                gridExtra_2.3              
 [91] compiler_3.6.1              biomaRt_2.40.5             
 [93] KernSmooth_2.23-17          crayon_1.3.4               
 [95] htmltools_0.5.0             mgcv_1.8-33                
 [97] later_1.0.0                 lubridate_1.7.9            
 [99] DBI_1.1.0                   dbplyr_1.4.4               
[101] MASS_7.3-53                 rappdirs_0.3.1             
[103] Matrix_1.2-18               cli_2.1.0                  
[105] igraph_1.2.5                pkgconfig_2.0.3            
[107] bigmemory.sri_0.1.3         GenomicAlignments_1.20.1   
[109] plotly_4.9.2.1              xml2_1.2.5                 
[111] foreach_1.5.1               XVector_0.24.0             
[113] rvest_0.3.5                 digest_0.6.25              
[115] sctransform_0.3.1           RcppAnnoy_0.0.16           
[117] graph_1.64.0                spatstat.data_1.4-3        
[119] Biostrings_2.52.0           cellranger_1.1.0           
[121] rmarkdown_2.1               leiden_0.3.3               
[123] uwot_0.1.8                  shiny_1.4.0.2              
[125] Rsamtools_2.0.3             gtools_3.8.2               
[127] lifecycle_0.2.0             nlme_3.1-150               
[129] jsonlite_1.7.1              fansi_0.4.1                
[131] viridisLite_0.3.0           pillar_1.4.3               
[133] lattice_0.20-41             fastmap_1.0.1              
[135] httr_1.4.1                  survival_3.2-7             
[137] glue_1.4.2                  spatstat_1.63-3            
[139] png_0.1-7                   iterators_1.0.13           
[141] bit_4.0.4                   stringi_1.5.3              
[143] blob_1.2.1                  caTools_1.18.0             
[145] memoise_1.1.0               irlba_2.3.3                
[147] future.apply_1.6.0         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
