<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sean Wilson" />

<meta name="date" content="2020-12-26" />

<title>Figures</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HowdenWilson2020</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="HFK_Annotation.html">HFK</a>
</li>
<li>
  <a href="BulkRNA-Seq.html">Bulk</a>
</li>
<li>
  <a href="Figures.html">Figures</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sbwilson91/HowdenWilson2020">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Figures</h1>
<h4 class="author">Sean Wilson</h4>
<h4 class="date">2020-12-26</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>HowdenWilson2020/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201109code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201109code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201109)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKidneyRegenerationHowdenWilson2020tree11477ded30366e99ed3566c3ae910e3fb4c1f47atargetblank11477dea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KidneyRegeneration/HowdenWilson2020/tree/11477ded30366e99ed3566c3ae910e3fb4c1f47a" target="_blank">11477de</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKidneyRegenerationHowdenWilson2020tree11477ded30366e99ed3566c3ae910e3fb4c1f47atargetblank11477dea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KidneyRegeneration/HowdenWilson2020/tree/11477ded30366e99ed3566c3ae910e3fb4c1f47a" target="_blank">11477de</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  README.html
    Untracked:  code/AddModuleScore_bulk.R
    Untracked:  code/GeneSets.Rmd
    Untracked:  code/functions.R
    Untracked:  code/hfk.R
    Untracked:  code/job_identity_markers.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Figures.Rmd</code>) and HTML (<code>docs/Figures.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/11477ded30366e99ed3566c3ae910e3fb4c1f47a/analysis/Figures.Rmd" target="_blank">11477de</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
<td>
Updating additional information around analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KidneyRegeneration/HowdenWilson2020/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/Figures.html" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/d1d8b3f0a1a0b90e15b3a664233c38d275476265/analysis/Figures.Rmd" target="_blank">d1d8b3f</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
<td>
Initialise website
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document shows the figures within the manuscript generated with R code.</p>
<pre class="r"><code>library(Seurat)
library(tidyverse)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;cli&#39;:
  method     from    
  print.boxx spatstat</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.0     ✓ purrr   0.3.4
✓ tibble  3.0.4     ✓ dplyr   1.0.2
✓ tidyr   1.0.2     ✓ stringr 1.4.0
✓ readr   1.4.0     ✓ forcats 0.5.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(NMF)</code></pre>
<pre><code>Loading required package: pkgmaker</code></pre>
<pre><code>Loading required package: registry</code></pre>
<pre><code>Loading required package: rngtools</code></pre>
<pre><code>Loading required package: cluster</code></pre>
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: synchronicity] | Cores 31/32</code></pre>
<pre><code>  To enable shared memory capabilities, try: install.extras(&#39;
NMF
&#39;)</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(patchwork)
source(here::here(&quot;code/functions.R&quot;))
source(here::here(&quot;code/AddModuleScore_bulk.R&quot;))
options(future.globals.maxSize = Inf)
# load the human fetal kidney data
load(file = here::here(&quot;data/Holloway2020.rda&quot;))
# load the ue/dn subsetted data
load(file = here::here(&quot;data/UE_DN_Subset.rda&quot;))</code></pre>
</div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p>Figure 1A:<br />
HFK single cell UMAP projection</p>
<pre class="r"><code>DimPlot(hfk, group.by = &quot;Identity&quot;, label = F, pt.size = 0.5,
              cols = rev(c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))))  +
  theme(
        axis.title = element_blank())</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-2-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-2-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 1B:<br />
Distal Nephron and Ureteric Epithelial cell types subset - UMAP projection</p>
<pre class="r"><code>DimPlot(ue.dn, group.by = &quot;IdentityRandom&quot;, label = T, repel = T, label.size = 4) +
  theme(legend.position = &quot;none&quot;,
        axis.title = element_blank()) </code></pre>
<pre><code>Warning: Using `as.character()` on a quosure is deprecated as of rlang 0.3.0.
Please use `as_label()` or `as_name()` instead.
This warning is displayed once per session.</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-3-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 1C:<br />
DotPlot of commonly used DN/UE markers</p>
<pre class="r"><code>ue.dn$Identity &lt;- factor(ue.dn$Identity,
                         levels = rev(c(&quot;N.Distal_EN&quot;, &quot;N.LoH&quot;, &quot;N.DST&quot;, &quot;N.DCT&quot;, &quot;N.CS&quot;,
                                        &quot;U.Tip&quot;, &quot;U.Cortical&quot;, &quot;U.Med_Outer&quot;, &quot;U.Med_Inner&quot;)))
ue.dn &lt;- SetIdent(ue.dn, value = &quot;Identity&quot;)
SWDP.col(ue.dn, features = rev(c(&quot;HOXB7&quot;, &quot;GATA3&quot;, &quot;CALB1&quot;, &quot;KRT8&quot;, &quot;KRT18&quot;, &quot;KRT19&quot;, &quot;AQP2&quot;,
                               &quot;CLU&quot;, &quot;MAL&quot;, &quot;SPP1&quot;, &quot;SPINT2&quot;, &quot;CDH16&quot;)), group.by = &quot;Identity&quot;, assay = &quot;RNA&quot;, col.max = 10) + theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>Warning: Removed 8 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-4-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 1D:<br />
DotPlot of more accurate DN/UE markers from Combes et al. 2019</p>
<pre class="r"><code>SWDP.col(ue.dn, features = rev(c(&quot;BSND&quot;, &quot;FXYD2&quot;, &quot;KCNJ1&quot;, &quot;KNG2&quot;, &quot;CLCNKB&quot;, &quot;CLEC18A&quot;,
                               &quot;KCNN4&quot;, &quot;AKR1B7&quot;, &quot;ALDH3B2&quot;, &quot;WNT9B&quot;, &quot;CTNND2&quot;, &quot;UPK1B&quot;, &quot;AQP2&quot;, &quot;RET&quot;)
                               ), group.by = &quot;Identity&quot;, assay = &quot;RNA&quot;, col.max = 10) +
  theme(legend.position = &quot;none&quot;) </code></pre>
<pre><code>Warning in FetchData(object = object, vars = features, cells = cells): The
following requested variables were not found: AKR1B7, KNG2</code></pre>
<pre><code>Warning: Removed 58 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-5-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-5-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 1E:<br />
DotPlot of the top 10 DE markers for each identity</p>
<pre class="r"><code>ue.dn.markers &lt;- FindAllMarkers(ue.dn, only.pos = T, verbose = T, test.use = &quot;t&quot;,
                                logfc.threshold = 0.25, min.pct = 0.5, min.diff.pct = 0.3)</code></pre>
<pre><code>Calculating cluster U.Med_Inner</code></pre>
<pre><code>Calculating cluster U.Med_Outer</code></pre>
<pre><code>Calculating cluster U.Cortical</code></pre>
<pre><code>Calculating cluster U.Tip</code></pre>
<pre><code>Calculating cluster N.CS</code></pre>
<pre><code>Calculating cluster N.DCT</code></pre>
<pre><code>Calculating cluster N.DST</code></pre>
<pre><code>Calculating cluster N.LoH</code></pre>
<pre><code>Calculating cluster N.Distal_EN</code></pre>
<pre class="r"><code>#ue.dn.markers$cluster &lt;- as.character(ue.dn.markers$cluster)

export.markers &lt;- lapply(1:length(unique(ue.dn.markers$cluster)), function(x) {
    ue.dn.markers %&gt;%
        dplyr::filter(cluster == unique(ue.dn.markers$cluster)[x], p_val_adj &lt; 0.05, avg_logFC &gt; 0) %&gt;%
        dplyr::arrange(-avg_logFC) %&gt;%
        transmute(Gene = gene, LogFC = avg_logFC, pVal = p_val_adj, pct.diff = pct.1-pct.2)
  
})


top10 &lt;- ue.dn.markers %&gt;% mutate(pct.diff = pct.1-pct.2) %&gt;% 
  group_by(cluster) %&gt;% top_n(10, wt = pct.diff) %&gt;% arrange(cluster, -pct.diff)
top5 &lt;- ue.dn.markers %&gt;% mutate(pct.diff = pct.1-pct.2) %&gt;% 
  group_by(cluster) %&gt;% top_n(5, wt = pct.diff) %&gt;% arrange(cluster, -pct.diff)</code></pre>
<pre class="r"><code>(SWDP.col(ue.dn, features = rev(unique(top10$gene[1:40])), group.by = &quot;Identity&quot;, assay = &quot;RNA&quot;, col.max = 10) +
  theme(legend.position = &quot;none&quot;)) /
(SWDP.col(ue.dn, features = rev(unique(top10$gene[41:90])), group.by = &quot;Identity&quot;, assay = &quot;RNA&quot;, col.max = 10))</code></pre>
<pre><code>Warning: Removed 159 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 108 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-7-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-7-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<p>Figure 2F:<br />
Heatmap showing the “Gene Set scores” of bulk RNA-seq. Samples are cell populations sorted from organoids (org) compared to a UB cultured population (UB).</p>
<pre class="r"><code>aheatmap(t(scores.all[rownames(scores.all)[19:30], ]),
         annCol = list(Sample = targets.new$Group[19:30]), annColors = list(c(&quot;red&quot;, &quot;orange&quot;, &quot;purple&quot;, &quot;green&quot;)), #annColors = list(pal.e),
         #color=vcol,
         #main=paste0(&quot;Gene set scores for
         #FACs organoid populations
         #and UB cultures&quot;), 
         Colv = NA,
         Rowv = NA,
         border = TRUE,
         scale = &quot;column&quot;, fontsize = 13)</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-9-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-9-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 2G:<br />
Heatmap showing expression of top differentially expressed genes between the UE and DN populations.</p>
<pre class="r"><code>aheatmap(t(v$E[v$genes$SYMBOL %in% genes$gene, 1:6]),
         annRow = list(Sample=c(&quot;ch+/Ep+ org&quot;, &quot;ch+/Ep+ org&quot;, &quot;ch+/Ep+ org&quot;, 
                                &quot;ch+/Ep+ UE&quot;, &quot;ch+/Ep+ UE&quot;, &quot;ch+/Ep+ UE&quot;)),
         annCol = list(Segment = geneorder$UEorDN), 
         annColors = list(c(&quot;violet&quot;, &quot;grey&quot;), c(&quot;purple&quot;, &quot;orange&quot;)), fontsize = 13, 
         border = TRUE, 
         legend = TRUE,
         labRow = colnames(v$E[1:6]),
         treeheight = c(10,25))</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-11-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-11-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>         #main=&quot;Expression of Genes from DN and UE&quot;, scale = &quot;none&quot;)</code></pre>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<pre class="r"><code>hfk.gata3 &lt;- hfk[, hfk$Identity %in% c(&quot;U.Tip&quot;, &quot;U.Cortical&quot;, &quot;U.Med_Inner&quot;, &quot;U.Med_Outer&quot;,
                                      &quot;N.DCT&quot;, &quot;N.CS&quot;, &quot;N.DistTubule_Dev&quot;, &quot;S.Mesangial&quot;)]
##hfk.gata3 &lt;- SCTransform(hfk.gata3, do.scale = T, verbose = F)
#ue.int &lt;- SCTransform(ue.int, do.scale = T, verbose = F)
#anchors.gata3 &lt;- FindTransferAnchors(hfk.gata3, query = ue.int, dims = 1:30, reference.assay = &quot;SCT&quot;, query.assay = &quot;SCT&quot;, #project.query = T)
#predictions &lt;- TransferData(anchorset = anchors.gata3, refdata = hfk.gata3$Identity, dims = 1:30)
#ue.int$gata3.seg.prediction &lt;- predictions$predicted.id
#ue.int$gata3.seg.max.score &lt;- predictions$prediction.score.max

calls &lt;- as.data.frame(table(ue.int$gata3.seg.prediction, ue.int$gata3.seg.max.score&gt;0.5,
                       ue.int$sample))
colnames(calls) &lt;- c(&quot;Predicted Segment&quot;, &quot;Similarity&quot;, &quot;Cells&quot;)
ue.int.df$pct &lt;- ue.int.df$Freq / sum(ue.int.df$Freq) * 100
colnames(ue.int.df) &lt;- c(&quot;Predicted Segment&quot;, &quot;Cells&quot;, &quot;Percent&quot;)

calls$Similarity &lt;- gsub(pattern = &quot;FALSE&quot;, x = calls$Similarity, replacement = &quot;Less than 50%&quot;)
calls$Similarity &lt;- gsub(pattern = &quot;TRUE&quot;, x = calls$Similarity, replacement = &quot;More than 50%&quot;)</code></pre>
<p>Figure 4A:<br />
UMAP representation of the two single cell UE samples</p>
<pre class="r"><code>DimPlot(ue.int, group.by = &quot;sample&quot;, ncol = 1) +
  #ggtitle(&quot;A. UE and DN segments transferred&quot;) +
    theme(legend.title=element_text(size=rel(0.5))) +
    theme(legend.text = element_text(size = rel(0.75)),
          axis.title = element_blank(),
          axis.text = element_blank())</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-13-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 4B:<br />
UMAP representatives of the two single cell UE samples, grouped by the outcome from “Label Transfer” with the human fetal kidney reference.</p>
<pre class="r"><code>DimPlot(ue.int, group.by = &quot;gata3.seg.prediction&quot;) +
  #ggtitle(&quot;A. UE and DN segments transferred&quot;) +
    theme(legend.title=element_text(size=rel(0.5))) +
    theme(legend.text = element_text(size = rel(0.75)),
          axis.title = element_blank(),
          axis.text = element_blank())</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-14-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 4C:<br />
Distribution of maximum similarity scores grouped by label transfer classification.</p>
<pre class="r"><code>predictions$predicted.id &lt;- gsub(&quot;N.DistTubule_Dev&quot;, &quot;N.DST&quot;, predictions$predicted.id)
predictions %&gt;% 
  mutate(`Predicted Segment` = predicted.id, `Max score` = prediction.score.max, 
         `Predicted Segment` = fct_relevel(`Predicted Segment`, 
            &quot;U.Tip&quot;, &quot;U.Cortical&quot;, &quot;U.Med_Outer&quot;, &quot;U.Med_Inner&quot;, &quot;N.CS&quot;, &quot;N.DST&quot;, &quot;S.Mesangial&quot;)) %&gt;%
  ggplot(aes(`Predicted Segment`, `Max score`)) +
    geom_jitter(aes(colour = predicted.id), size = 0.05, alpha = 0.8)+
  geom_violin(alpha = 0.5) +
  theme_light()+
  scale_y_continuous(limits = c(0,1)) +
    theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust = 0, size = 20)) +
    theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>Warning: Removed 46 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 46 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-15-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-15-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 4D:<br />
UMAP plots showing the expression of important UE and DN genes in the single cell UE culture samples.</p>
<pre class="r"><code>  #ggtitle(&quot;Max similarity per cell&quot;)
genes &lt;- c(&quot;RET&quot;, &quot;WNT11&quot;, 
           &quot;WNT9B&quot;, &quot;WNT7B&quot;,
           &quot;AQP2&quot;, &quot;UPK2&quot;)
lp &lt;- list()
for (gene in genes) {
  p &lt;- FeaturePlot(ue.int, order = T,
            features =  c(paste0(gene)), pt.size = 0.3, 
                  cols = c(vcol)) + 
  theme(axis.title = element_blank(),
        plot.title = element_text(size = 12),
        legend.text = element_blank(),
        axis.text = element_blank(),
        legend.position = &quot;none&quot;)
  lp[[paste0(gene)]] &lt;- p
}

lp %&gt;%  patchwork::wrap_plots(design = &quot;
                                 ACF
                                 BDE&quot;)</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-16-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-7" class="section level2">
<h2>Figure 7</h2>
<p>Figure 7A:<br />
Heatmap showing expression of interesting genes for UE and Stalk in bulk RNA-seq samples of cultured UE and Stalk.</p>
<pre class="r"><code>targets.new &lt;- read_csv(here::here(&quot;data/AllTargets.csv&quot;))</code></pre>
<pre><code>
── Column specification ────────────────────────────────────────────────────────
cols(
  Sample = col_character(),
  `Sequencing Name` = col_character(),
  `Replicate wells` = col_double(),
  Experiment = col_character(),
  Age = col_character(),
  Treatment = col_character(),
  `Date sample harvested` = col_character(),
  `Date RNA extracted` = col_character(),
  `RNA conc (ng/uL)` = col_double(),
  `Volume (uL)` = col_double(),
  `Total RNA (ng)` = col_double(),
  Group = col_character(),
  Rep = col_double(),
  Batch = col_character(),
  ID = col_character(),
  Exp.Origin = col_character()
)</code></pre>
<pre class="r"><code>batch &lt;- ggplotColors(length(unique(targets.new$Batch)))
treatment &lt;- RColorBrewer::brewer.pal(3, &quot;Set2&quot;)

#gene &lt;- RColorBrewer::brewer.pal(length(unique(gene.markers$gene)), &quot;Set3&quot;)
load(file = here::here(&quot;data/Voom_Stalk.rda&quot;))

gene.markers &lt;- tibble(gene = c(&quot;AQP2&quot;, &quot;SPINK1&quot;, &quot;SLC26A4&quot;, &quot;ATP6V1B1&quot;, &quot;WNT9B&quot;, &quot;SLC4A5&quot;, &quot;OXGR1&quot;, &quot;AVPR1A&quot;, &quot;AVPR2&quot;, &quot;WNT7B&quot;, &quot;SLC26A7&quot;, &quot;SCNN1G&quot;, &quot;CALB1&quot;, &quot;RHBG&quot;, &quot;WNT11&quot;),
                       segment = c(&quot;PC&quot;, &quot;IC&quot;,    &quot;IC&quot;,      &quot;IC&quot;,       &quot;Stalk&quot;, &quot;Stalk&quot;,  &quot;IC&quot;,    &quot;IC&quot;,     &quot;PC&quot;,    &quot;Stalk&quot;, &quot;IC&quot;,      &quot;PC&quot;,     &quot;PC&quot;,     &quot;Stalk&quot;, &quot;Stalk&quot;),
                       id = 1:15)
segment &lt;- RColorBrewer::brewer.pal(length(unique(gene.markers$segment)), &quot;Set1&quot;)</code></pre>
<pre class="r"><code>aheatmap((v2$E[v2$genes$SYMBOL %in% gene.markers$gene,
              c(16:21, 31:33)]),
         annCol = list(Treatment=targets.new$Treatment[c(16:21, 31:33)]),
         annColors = list(Treatment = treatment), 
         fontsize = 13, 
         border = TRUE, 
         treeheight = c(10,25),
         main=&quot;&quot;, scale = &quot;none&quot;)</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KidneyRegeneration/HowdenWilson2020/blob/1297fee0a7cfbef1dce2b0410d4edd3c1d2c1190/docs/figure/Figures.Rmd/unnamed-chunk-18-1.png" target="_blank">1297fee</a>
</td>
<td>
Sean Wilson
</td>
<td>
2020-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Figure 7B:<br />
Heatmap showing Gene Set profiles in bulk RNA-seq samples of cultured UE and Stalk.</p>
<pre class="r"><code>aheatmap(t(scores.all)[16:27, c(16:21, 31:33)],
         annCol = list(Treatment=targets.new$Treatment[c(16:21, 31:33)]),
         annColors = list(Treatment = treatment), 
         fontsize = 13, 
         border = TRUE, 
         treeheight = c(10,25),
         main=&quot;&quot;, scale = &quot;none&quot;)</code></pre>
<p><img src="figure/Figures.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /hpc/software/installed/R/3.6.1/lib64/R/lib/libRblas.so
LAPACK: /hpc/software/installed/R/3.6.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] limma_3.42.2        patchwork_1.0.0     RColorBrewer_1.1-2 
 [4] NMF_0.22.0          bigmemory_4.5.36    Biobase_2.46.0     
 [7] BiocGenerics_0.32.0 cluster_2.1.0       rngtools_1.5       
[10] pkgmaker_0.32.2     registry_0.5-1      forcats_0.5.0      
[13] stringr_1.4.0       dplyr_1.0.2         purrr_0.3.4        
[16] readr_1.4.0         tidyr_1.0.2         tibble_3.0.4       
[19] ggplot2_3.3.0       tidyverse_1.3.0     Seurat_3.2.2       
[22] workflowr_1.6.2    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1          backports_1.1.10      plyr_1.8.6           
  [4] igraph_1.2.5          lazyeval_0.2.2        splines_3.6.1        
  [7] listenv_0.8.0         gridBase_0.4-7        digest_0.6.25        
 [10] foreach_1.5.1         htmltools_0.5.0       viridis_0.5.1        
 [13] fansi_0.4.1           magrittr_1.5          tensor_1.5           
 [16] doParallel_1.0.16     ROCR_1.0-11           globals_0.13.0       
 [19] modelr_0.1.6          matrixStats_0.57.0    colorspace_1.4-1     
 [22] blob_1.2.1            rvest_0.3.5           rappdirs_0.3.1       
 [25] ggrepel_0.8.2         haven_2.2.0           xfun_0.12            
 [28] bigmemory.sri_0.1.3   crayon_1.3.4          jsonlite_1.7.1       
 [31] spatstat_1.63-3       spatstat.data_1.4-3   survival_3.2-7       
 [34] zoo_1.8-8             iterators_1.0.13      glue_1.4.2           
 [37] polyclip_1.10-0       gtable_0.3.0          leiden_0.3.3         
 [40] future.apply_1.6.0    abind_1.4-5           scales_1.1.1         
 [43] DBI_1.1.0             miniUI_0.1.1.1        Rcpp_1.0.5           
 [46] viridisLite_0.3.0     xtable_1.8-4          reticulate_1.16      
 [49] rsvd_1.0.3            htmlwidgets_1.5.1     httr_1.4.1           
 [52] ellipsis_0.3.1        ica_1.0-2             farver_2.0.3         
 [55] pkgconfig_2.0.3       uwot_0.1.8            dbplyr_1.4.4         
 [58] deldir_0.1-29         here_0.1              labeling_0.4.2       
 [61] tidyselect_1.1.0      rlang_0.4.7           reshape2_1.4.4       
 [64] later_1.0.0           munsell_0.5.0         cellranger_1.1.0     
 [67] tools_3.6.1           cli_2.1.0             generics_0.1.0       
 [70] broom_0.7.2           ggridges_0.5.2        evaluate_0.14        
 [73] fastmap_1.0.1         yaml_2.2.1            goftest_1.2-2        
 [76] knitr_1.28            fs_1.5.0              fitdistrplus_1.1-1   
 [79] RANN_2.6.1            pbapply_1.4-3         future_1.16.0        
 [82] nlme_3.1-150          whisker_0.4           mime_0.9             
 [85] xml2_1.2.5            compiler_3.6.1        rstudioapi_0.11      
 [88] plotly_4.9.2.1        png_0.1-7             spatstat.utils_1.17-0
 [91] reprex_0.3.0          stringi_1.5.3         lattice_0.20-41      
 [94] Matrix_1.2-18         vctrs_0.3.4           pillar_1.4.3         
 [97] lifecycle_0.2.0       lmtest_0.9-38         RcppAnnoy_0.0.16     
[100] data.table_1.13.0     cowplot_1.1.0         irlba_2.3.3          
[103] httpuv_1.5.2          R6_2.5.0              promises_1.1.0       
[106] KernSmooth_2.23-17    gridExtra_2.3         codetools_0.2-16     
[109] MASS_7.3-53           assertthat_0.2.1      rprojroot_1.3-2      
[112] withr_2.2.0           sctransform_0.3.1     mgcv_1.8-33          
[115] hms_0.5.3             grid_3.6.1            rpart_4.1-15         
[118] rmarkdown_2.1         Rtsne_0.15            git2r_0.27.1         
[121] shiny_1.4.0.2         lubridate_1.7.9      </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /hpc/software/installed/R/3.6.1/lib64/R/lib/libRblas.so
LAPACK: /hpc/software/installed/R/3.6.1/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] limma_3.42.2        patchwork_1.0.0     RColorBrewer_1.1-2 
 [4] NMF_0.22.0          bigmemory_4.5.36    Biobase_2.46.0     
 [7] BiocGenerics_0.32.0 cluster_2.1.0       rngtools_1.5       
[10] pkgmaker_0.32.2     registry_0.5-1      forcats_0.5.0      
[13] stringr_1.4.0       dplyr_1.0.2         purrr_0.3.4        
[16] readr_1.4.0         tidyr_1.0.2         tibble_3.0.4       
[19] ggplot2_3.3.0       tidyverse_1.3.0     Seurat_3.2.2       
[22] workflowr_1.6.2    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1          backports_1.1.10      plyr_1.8.6           
  [4] igraph_1.2.5          lazyeval_0.2.2        splines_3.6.1        
  [7] listenv_0.8.0         gridBase_0.4-7        digest_0.6.25        
 [10] foreach_1.5.1         htmltools_0.5.0       viridis_0.5.1        
 [13] fansi_0.4.1           magrittr_1.5          tensor_1.5           
 [16] doParallel_1.0.16     ROCR_1.0-11           globals_0.13.0       
 [19] modelr_0.1.6          matrixStats_0.57.0    colorspace_1.4-1     
 [22] blob_1.2.1            rvest_0.3.5           rappdirs_0.3.1       
 [25] ggrepel_0.8.2         haven_2.2.0           xfun_0.12            
 [28] bigmemory.sri_0.1.3   crayon_1.3.4          jsonlite_1.7.1       
 [31] spatstat_1.63-3       spatstat.data_1.4-3   survival_3.2-7       
 [34] zoo_1.8-8             iterators_1.0.13      glue_1.4.2           
 [37] polyclip_1.10-0       gtable_0.3.0          leiden_0.3.3         
 [40] future.apply_1.6.0    abind_1.4-5           scales_1.1.1         
 [43] DBI_1.1.0             miniUI_0.1.1.1        Rcpp_1.0.5           
 [46] viridisLite_0.3.0     xtable_1.8-4          reticulate_1.16      
 [49] rsvd_1.0.3            htmlwidgets_1.5.1     httr_1.4.1           
 [52] ellipsis_0.3.1        ica_1.0-2             farver_2.0.3         
 [55] pkgconfig_2.0.3       uwot_0.1.8            dbplyr_1.4.4         
 [58] deldir_0.1-29         here_0.1              labeling_0.4.2       
 [61] tidyselect_1.1.0      rlang_0.4.7           reshape2_1.4.4       
 [64] later_1.0.0           munsell_0.5.0         cellranger_1.1.0     
 [67] tools_3.6.1           cli_2.1.0             generics_0.1.0       
 [70] broom_0.7.2           ggridges_0.5.2        evaluate_0.14        
 [73] fastmap_1.0.1         yaml_2.2.1            goftest_1.2-2        
 [76] knitr_1.28            fs_1.5.0              fitdistrplus_1.1-1   
 [79] RANN_2.6.1            pbapply_1.4-3         future_1.16.0        
 [82] nlme_3.1-150          whisker_0.4           mime_0.9             
 [85] xml2_1.2.5            compiler_3.6.1        rstudioapi_0.11      
 [88] plotly_4.9.2.1        png_0.1-7             spatstat.utils_1.17-0
 [91] reprex_0.3.0          stringi_1.5.3         lattice_0.20-41      
 [94] Matrix_1.2-18         vctrs_0.3.4           pillar_1.4.3         
 [97] lifecycle_0.2.0       lmtest_0.9-38         RcppAnnoy_0.0.16     
[100] data.table_1.13.0     cowplot_1.1.0         irlba_2.3.3          
[103] httpuv_1.5.2          R6_2.5.0              promises_1.1.0       
[106] KernSmooth_2.23-17    gridExtra_2.3         codetools_0.2-16     
[109] MASS_7.3-53           assertthat_0.2.1      rprojroot_1.3-2      
[112] withr_2.2.0           sctransform_0.3.1     mgcv_1.8-33          
[115] hms_0.5.3             grid_3.6.1            rpart_4.1-15         
[118] rmarkdown_2.1         Rtsne_0.15            git2r_0.27.1         
[121] shiny_1.4.0.2         lubridate_1.7.9      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
